File: MinecraftAdventure.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        String input = reader.readLine();
        int numSticks = 0, numStone = 0;
        while (!input.equals("END")){
            if (input.equals("Sticks")){
                numSticks++;
            }
            if (input.equals("Wood")) {
                numSticks += 4;
            }
            if (input.equals("Cobblestone")) {
                numStone++;
            }
            input = reader.readLine();
        }
        int amount = 0;
        if (numSticks >= 2 && numStone >= 3){
            numSticks /= 2;
            numStone /= 3;
            amount = Math.min(numSticks, numStone);
        }
        System.out.println("Amount of stone pickaxes: " + amount);
    }
}

File: minecraft_adventure.rs
use std::io;
fn main() {
    let mut num_sticks = 0;
    let mut num_stone = 0;
    loop {
        let mut input = String::new();
        io::stdin().read_line(&mut input).unwrap();
        let input = input.trim().to_string();
        if input == "Sticks" {
            num_sticks += 1;
        }
        if input == "Wood" {
            num_sticks += 4;
        }
        if input == "Stone" {
            num_stone += 1;
        }
        if input == "END" {
            break;
        }
    }
    if num_sticks >= 2 && num_stone >= 3 {
        num_sticks /= 2;
        num_stone /= 3;
        println!("{}", if num_sticks < num_stone {num_sticks} else {num_stone})
    } else {
        println!("{}", 0)
    }
}
